wallet/src/multisig_account.rs

```diff
@@ -0,0 +1,232 @@
+        &self,
+        recipient: Address,
+        value: Coin,
+        fee: Coin,
+        validity_start_height: u32,
+        network_id: NetworkId,
+    ) -> Transaction {
+        Transaction::new_basic(
+            self.address.clone(),
+            recipient,
+            value,
+            fee,
+            validity_start_height,
+            network_id,
+        )
+    }
+
+    /// Utility method that delinearizes and aggregates the provided slice of public keys.
+    pub fn aggregate_public_keys(public_keys: &[PublicKey]) -> PublicKey {
+        let mut sorted_public_keys = Vec::with_capacity(public_keys.len());
+        public_keys.clone_into(&mut sorted_public_keys);
+        sorted_public_keys.sort();
+
+        let public_keys_hash = hash_public_keys(&sorted_public_keys);
+        let delinearized_pk_sum: EdwardsPoint = public_keys
+            .iter()
+            .map(|public_key| public_key.delinearize(&public_keys_hash))
+            .sum();
+
+        let mut public_key_bytes: [u8; PublicKey::SIZE] = [0u8; PublicKey::SIZE];
+        public_key_bytes.copy_from_slice(delinearized_pk_sum.compress().as_bytes());
+        PublicKey::from(public_key_bytes)
+    }
+
+    /// Creates a partial signature of the provided transaction.
+    pub fn partially_sign_transaction(
+        &self,
+        transaction: &Transaction,
+        public_keys: &[PublicKey],
+        commitments: &[Commitment],
+        secret: &RandomSecret,
+    ) -> PartialSignature {
+        let mut sorted_public_keys = Vec::with_capacity(public_keys.len());
+        public_keys.clone_into(&mut sorted_public_keys);
+        sorted_public_keys.sort();
+
+        self.key_pair
+            .partial_sign(
+                &sorted_public_keys,
+                secret,
+                commitments,
+                transaction.serialize_content().as_slice(),
+            )
+            .0
+    }
+
+    /// Creates a signature proof. Panics if the length of `signatures` is not equal to `min_signatures`.
+    pub fn create_proof(
+        &self,
+        aggregated_public_key: &PublicKey,
+        aggregated_commitment: &Commitment,
+        signatures: &[PartialSignature],
+    ) -> SignatureProof {
+        assert!(
+            signatures.len() == self.min_signatures as usize,
+            "Number of signatures must be the same as the minimal signatures."
+        );
+
+        let aggregated_signature: PartialSignature = signatures.iter().sum();
+        let raw_aggregated_signature = aggregated_signature.as_bytes();
+        let raw_aggregated_commitment = aggregated_commitment.to_bytes();
+
+        let mut combined =
+            Vec::with_capacity(raw_aggregated_signature.len() + raw_aggregated_commitment.len());
+        combined.extend_from_slice(&aggregated_commitment.to_bytes());
+        combined.extend_from_slice(aggregated_signature.as_bytes());
+        let signature = Signature::from_bytes(&combined).expect("Invalid signature constructued.");
```

Comment:
If we return a `None` or error, we should remove this `expect` and convert it to the appropriate return type